e2e/tests/test_smoke.py

```diff
@@ -0,0 +1,191 @@
+                    }
+                 }
+            }
+        }
+        k8s_api.apps_v1.patch_namespaced_deployment("postgres-operator", "default", body)
+
+        # TODO check if CRD is registered instead
+        Utils.wait_for_pod_start(k8s_api, 'name=postgres-operator', cls.RETRY_TIMEOUT_SEC)
+
+        # HACK around the lack of Python client for the acid.zalan.do resource
+        subprocess.run(["kubectl", "create", "-f", "manifests/minimal-postgres-manifest.yaml"])
+
+        Utils.wait_for_pod_start(k8s_api, 'spilo-role=master', cls.RETRY_TIMEOUT_SEC)
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_master_is_unique(self):
+        """
+           Check that there is a single pod in the k8s cluster with the label "spilo-role=master".
+        """
+        k8s = K8sApi()
+        labels = 'spilo-role=master,version=acid-minimal-cluster'
+
+        num_of_master_pods = Utils.count_pods_with_label(k8s, labels)
+        self.assertEqual(num_of_master_pods, 1, "Expected 1 master pod, found {}".format(num_of_master_pods))
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_scaling(self):
+        """
+           Scale up from 2 to 3 pods and back to 2 by updating the Postgres manifest at runtime.
+        """
+        k8s = K8sApi()
+        labels = "version=acid-minimal-cluster"
+
+        Utils.wait_for_pg_to_scale(k8s, 3, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(3, Utils.count_pods_with_label(k8s, labels))
+
+        # TODO `kind` pods may stuck in the Terminating state for a few minutes; reproduce/file a bug report
+        Utils.wait_for_pg_to_scale(k8s, 2, self.RETRY_TIMEOUT_SEC)
+        self.assertEqual(2, Utils.count_pods_with_label(k8s, labels))
+
+    @timeout_decorator.timeout(TEST_TIMEOUT_SEC)
+    def test_taint_based_eviction(self):
+        """
+           Add taint "postgres=:NoExecute" to node with master.
+        """
+        k8s = K8sApi()
+        labels = 'version=acid-minimal-cluster'
+        master_pod_node = ''
+        new_master_pod_node = ''
+
+        body = {
+            "spec": {
+                "taints": [
+                    {
+                        "effect": "NoExecute",
+                        "key": "postgres"
+                    }
+                ]
+            }
+        }
+        podsList = k8s.core_v1.list_namespaced_pod("default", label_selector=labels)
+        for pod in podsList.items:
+            if ('spilo-role', 'master') in pod.metadata.labels.items():
+                master_pod_node = pod.spec.node_name
+            elif ('spilo-role', 'replica') in pod.metadata.labels.items():
+                new_master_pod_node = pod.spec.node_name
+
+        k8s.core_v1.patch_node(master_pod_node, body)
+        Utils.wait_for_master_failover(k8s, new_master_pod_node, self.RETRY_TIMEOUT_SEC)
+
+        self.assertTrue(master_pod_node != new_master_pod_node, "Master on {} did not fail over to {}".format(master_pod_node, new_master_pod_node))
```

Comment:
why is the `new_master_pod_node` not updated after failover ? at this point we still need to check that whatever runs on   
`new_master_pod_node` is actually the master of a given cluster. Otherwise assertion is redundant